src/index.ts

```diff
@@ -32,41 +51,124 @@ export const options = {
       // Start `rate` iterations per second
       timeUnit: "1s",
 
-      // Pre-allocate VUs
-      preAllocatedVUs: 2,
+      // Pre-allocate VUs (concurrent users)
+      preAllocatedVUs: keys.length,
     },
   },
 };
 
-const config = getConfigOrThrow();
+const generateNonceDuration = new Trend("generate_nonce_duration");
+const refreshFastLoginDuration = new Trend("fast_login_duration");
+const refreshFastLoginWaiting = new Trend("fast_login_waiting");
+const sessionDuration = new Trend("get_session_duration");
+const scenarioDuration = new Trend("scenario_duration");
 
-export default function() {
+export default async function() {
+  let duration = 0;
   // Generate Nonce
-  const response = http.post(
-    `${config.IO_BACKEND_BASE_URL}/fast-login/nonce/generate`,
+  const generateNonceResponse = http.post(
+    `${config.IO_BACKEND_BASE_URL}/api/v1/fast-login/nonce/generate`,
     undefined,
-    { responseType: "text" }
+    {
+      responseType: "text",
+    }
   );
-  check(response, {
+  check(generateNonceResponse, {
     "GET Nonce returns 200": (r) => r.status === 200,
   });
-  pipe(
-    response.json(),
+  generateNonceDuration.add(generateNonceResponse.timings.duration);
+  duration += generateNonceResponse.timings.duration;
+  const nonce = pipe(
+    generateNonceResponse.json(),
     GenerateNonceResponse.decode,
     E.map((_) => _.nonce),
     E.getOrElseW((_) => {
       console.error("Error decoding nonce");
       fail(readableReportSimplified(_));
     })
   );
-  console.log(`${exec.vu.idInInstance}`);
-  // Build sign request (local)
-  // Refresh
-  // Get Session
+
+  // Generate Signature params for lollipop
+  const parameters = {
+    privateKeyJwk: JSON.stringify(keys[exec.vu.idInInstance - 1].privateKey),
+    thumbprint: keys[exec.vu.idInInstance - 1].thumbprint,
+    nonce,
+    url: config.IO_BACKEND_BASE_URL + "/api/v1/fast-login",
+  };
+  const signerResponse = http.post(
+    `http://backend.localhost:8001/signature-params`,
```

Comment:
may I suggest this change?
with this the user won't need to bind the address to `/etc/hosts`
```suggestion
    `http://localhost:8001/signature-params`,
```